app/user/user_exceptions.c

```diff
@@ -73,13 +72,15 @@ void load_non_32_wide_handler (struct exception_frame *ef, uint32_t cause)
   {
 die:
     /* Turns out we couldn't fix this, so try and chain to the handler
-     * that was set by the SDK. If none then trigger a system break instead
-     * and hang if the break doesn't get handled. This is effectively
-     * what would happen if the default handler was installed. */
+     * that was set. (This is typically a remote GDB break). If none 
+     * then trigger a system break instead and hang if the break doesn't 
+     * get handled. This is effectively what would happen if the default 
+     * handler was installed. */
     if (load_store_handler) {
       load_store_handler(ef, cause);
+      return;
     }
-    asm ("break 1, 1");
+     asm ("break 1, 1");
```

Comment:
Surplus whitespace, or intentional?

Reply:
Surplus

Reply:
I'd lose the while(1) {} and replace it by a while  around the break.  But a job for another day.  

The break will instantly reboot the ESP if the debugger exception handler is not installed If is then you will enter the debugger.